modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/APIApplicationRelevantObjectEntryModelListener.java

```diff
@@ -62,6 +92,61 @@ public void onBeforeUpdate(
 		_validate(objectEntry);
 	}
 
+	private void _manageAPIApplication(ObjectEntry objectEntry)
+		throws Exception {
+
+		Map<String, Serializable> values = objectEntry.getValues();
+
+		APIApplication apiApplication =
+			_apiApplicationProvider.fetchAPIApplication(
+				(String)values.get("baseURL"),
+				CompanyThreadLocal.getCompanyId());
```

Comment:
Could we get the companyId from the ObjectEntry?

Reply:
You are right @ccorreagg!! I will change it